[EDITS}W-11957883-VPN-risk-editorial=review (#649)
mulesoft/docs-runtime-manager: cloudhub-1/modules/ROOT/Edits/vpn-maintenance.adoc

= Anypoint VPN Maintenance
ifndef::env-site,env-github[]
include::_attributes.adoc[]
endif::[]
:page-aliases: runtime-manager::vpc-maintenance.adoc,

== Anypoint VPN Update

> MuleSoft Anypoint VPN is a managed service that is underpinned by AWS Site-to-Site VPN connections. VPN is key for any integrations or flows that require connectivity to your on-premises or other external platforms. Updating an Anypoint VPN that is not set up with at least a pair of tunnels using BGP routing will cause downtime. See xref:vpn-high-availability.adoc[Anypoint VPN High Availability] for more information.

Reviewer: ```suggestion
MuleSoft Anypoint VPN is a managed service that is underpinned by AWS Site-to-Site VPN connections. Using a VPN is important for any integrations or flows that require connectivity to your on-premises or other external platforms. Updating an Anypoint VPN that is not set up with at least a pair of tunnels that use BGP routing will cause downtime. See xref:vpn-high-availability.adoc[Anypoint VPN High Availability] for more information.
```

> MuleSoft Anypoint VPN is a managed service that is underpinned by AWS Site-to-Site VPN connections. Using a VPN is important for any integrations or flows that require connectivity to your on-premises or other external platforms. Updating an Anypoint VPN that is not set up with at least a pair of tunnels that use BGP routing will cause downtime. See xref:vpn-high-availability.adoc[Anypoint VPN High Availability] for more information.

Reviewer: I think you can remove "other" because it implies that the on-premises platform is external.

After the October 2022 quarterly maintenance window, Anypoint VPN will no longer support quarterly customer self-upgrades. From this point forward, updates made to Anypoint VPN will adhere to AWS's standard continuous upgrade process. Upgrades will be applied on-demand, with no published maintenance window schedule, no customer opt-in self-upgrade capabilities, with *potential (limited) downtime for customers that do not have a* xref:vpn-high-availability.adoc[High Availability (HA) VPN implementation].

> Deploy a xref:vpn-high-availability.adoc[high-availability VPN] solution to avoid any downtime associated with these AWS-delivered security updates. Review the below section for recommendations and troubleshooting.

Reviewer: ```suggestion
Deploy a xref:vpn-high-availability.adoc[high-availability VPN] solution to avoid any downtime associated with these AWS-delivered security updates. Review the following sections for recommendations and troubleshooting information.
```

== Avoiding or Reducing Downtime During an On-demand Maintenance

> Mule applications that use Anypoint VPN connectivity are impacted during the on-demand maintenance. The impact on the Anypoint VPN varies depending on the VPN configuration:

Reviewer: ```suggestion
Mule applications that use Anypoint VPN connectivity are impacted during an on-demand maintenance. The impact on the Anypoint VPN varies depending on the VPN configuration:
```

> * An Anypoint VPN with at least one pair of tunnels configured with BGP routing and DPD (Dead Peer Detection) to reset the tunnels when triggered incurs minimal downtime. The switchover takes a few seconds. See https://help.mulesoft.com/s/article/DPD-Dead-Peer-Detection-and-Anypoint-VPN-Virtual-Private-Networking[this article^] for an explanation of DPD and VPN.

Reviewer: Consider naming the article in the link.

> * An Anypoint VPN tunnel pair configured with Static routing incurs some downtime before reconnection. The reconnection takes longer if you do not have an automated failover mechanism to detect failed tunnels.

Reviewer: ```suggestion
* An Anypoint VPN tunnel pair configured with static routing incurs some downtime before reconnection. The reconnection takes longer if you do not have an automated failover mechanism to detect failed tunnels.
```

* For an Anypoint VPN configured without BGP routing or a failover mechanism or with only one tunnel configured, the outage is up to 60 minutes long while the update occurs and the original tunnel comes back up.

== Troubleshoot Issues During an Anypoint VPN On-demand Maintenance

> These are some of the possible issues you may encounter during the On-demand Maintenance:

Reviewer: ```suggestion
These are some of the possible issues you may encounter during an on-demand maintenance:
```

> === Tunnels not Coming Up

Reviewer: ```suggestion
=== Tunnels Don't Come Up
```

If the tunnels are not back up within 10-15 minutes at any point in time, try restarting your end of the tunnels to bring them back up.

> If your VPN is configured with static routing, try generating 'interesting' traffic towards the Anypoint VPN to bring up the VPN. See https://help.mulesoft.com/s/article/How-to-Generate-Interesting-Traffic-for-Anypoint-VPN[How to Generate Interesting Traffic for Anypoint VPN^] for more information.

Reviewer: Did you mean to put 'interesting' in backticks instead of single quotes? If you want to leave it in quotes, I would use double quotes.

> === Tunnel is Going Down Due to Inactivity

Reviewer: ```suggestion
=== Tunnel Goes Down Due to Inactivity
```

> If there is no traffic over the VPN tunnel, it goes down due to inactivity. In this case, generate interesting traffic. We recommend you move towards BGP.

Reviewer: ```suggestion
If there is no traffic over the VPN tunnel, the VPN goes down due to inactivity. In this case, generate interesting traffic. We recommend that you move towards BGP.
```

Reviewer: Put the word "interesting" in backticks or quotes for consistency with other uses of the elsewhere.

=== IKE Phase Lifetimes Mismatch

> ==== If You Are Using IKEv2:

Reviewer: ```suggestion
==== If You Use IKEv2:
```

> To maintain a reliable tunnel operation after the on-demand maintenance, ensure that IKEv2 Phase 1 and Phase 2 lifetimes are set as described in xref:vpn-about.adoc#ipsec-settings[the documentation]:

Reviewer: ```suggestion
To maintain a reliable tunnel operation after an on-demand maintenance, ensure that the IKEv2 Phase 1 and Phase 2 lifetimes are set as described in xref:vpn-about.adoc#ipsec-settings[the documentation]:
```

Reviewer: Can you name the topic in the link instead of saying "the documentation"?

* IKE Phase 1 lifetime: 28000 seconds
* IKE Phase 2 lifetime: 3000 second

[NOTE]

> AWS documentation states 28800 seconds for IKE Phase 1 lifetime, and 3600 seconds for IKE Phase 2 lifetime. MuleSoft recommends the 28000/3000 combination for your firewall device to timeout first and control the rekey process.

Reviewer: ```suggestion
AWS documentation states 28800 seconds for the IKE Phase 1 lifetime, and 3600 seconds for the IKE Phase 2 lifetime. MuleSoft recommends the 28000/3000 combination for your firewall device to timeout first and control the rekey process.
```

> ==== If You Are Using IKEv1:

Reviewer: ```suggestion
==== If You Use IKEv1:
```

> If you are using IKEv1 please use the default configuration: 28800 seconds for Phase 1 lifetime, and 3600 seconds for Phase 2 lifetime.

Reviewer: ```suggestion
If you use IKEv1, use the default configuration: 28800 seconds for the Phase 1 lifetime, and 3600 seconds for the Phase 2 lifetime.
```

=== Anypoint VPN Goes Down Periodically Due to PFS not Enabled for IPsec Phase 2

> If you are using Azure VPN, ensure that PFS is enabled and adjust lifetimes for Phase 2. For more information, see:

Reviewer: ```suggestion
If you are use Azure VPN, ensure that PFS is enabled and adjust the lifetimes for Phase 2. For more information, see:
```

> If you are use Azure VPN, ensure that PFS is enabled and adjust the lifetimes for Phase 2. For more information, see:

Reviewer: ```suggestion
If you are Azure VPN, ensure that PFS is enabled and adjust the lifetimes for Phase 2. For more information, see:
```

* https://help.mulesoft.com/s/article/Anypoint-VPN-goes-Down-Periodically-Due-to-PFS-is-not-Enabled-for-IPsec-Phase-2[Anypoint VPN Goes Down Periodically Due to PFS not Enabled for IPsec Phase 2^]
* https://help.mulesoft.com/s/article/Anypoint-VPN-to-Azure-keeps-going-down-every-hour[Anypoint VPN to Azure Keeps Going Down Every Hour^]

=== Anypoint VPN with BGP Routing Enters Asymmetric Mode after Update

> Anypoint VPN enters into asymmetric model after an on-demand maintenance if the AS paths for the two tunnels of the same VPN are different. See https://help.mulesoft.com/s/article/Anypoint-VPN-with-BGP-Routing-Enters-Asymmetric-Mode-After-update[Anypoint VPN with BGP Routing Enters Asymmetric Mode after Upgrade^] for more information.

Reviewer: asymmetric model > an asymmetric mode

I think you mean "mode" instead of "model".
.

Author: Yep. Checked with my PM, and it’s "mode"

=== Seeing IPsec DPD Failures in VPN Logs

> Ensure your DPD settings are correctly configured. See https://help.mulesoft.com/s/article/Seeing-IPsec-DPD-Failures-in-VPN-Logs[Seeing IPsec DPD Failures in VPN Logs^] for more information.

Reviewer: ```suggestion
Ensure that your DPD settings are correctly configured. See https://help.mulesoft.com/s/article/Seeing-IPsec-DPD-Failures-in-VPN-Logs[Seeing IPsec DPD Failures in VPN Logs^] for more information.
```

=== Other issues

> For other issues with Anypoint VPN not particularly related to the on-demand maintenance itself, see the general troubleshooting xref:vpn-troubleshooting.adoc[documentation].

Reviewer: ```suggestion
For other issues with Anypoint VPN, see the general troubleshooting xref:vpn-troubleshooting.adoc[documentation].
```
